v-next/hardhat-ignition/src/internal/tasks/deploy.ts

```diff
@@ -0,0 +1,337 @@
+import type { HardhatRuntimeEnvironment } from "@ignored/hardhat-vnext/types/hre";
+import type { NewTaskActionFunction } from "@ignored/hardhat-vnext/types/tasks";
+import type {
+  DeploymentParameters,
+  DeploymentResult,
+} from "@ignored/hardhat-vnext-ignition-core";
+
+import path from "node:path";
+
+import { HardhatError } from "@ignored/hardhat-vnext-errors";
+import { deploy, IgnitionError } from "@ignored/hardhat-vnext-ignition-core";
+import {
+  ensureDir,
+  exists,
+  readdir,
+  readJsonFile,
+  remove,
+  writeJsonFile,
+} from "@ignored/hardhat-vnext-utils/fs";
+import debug from "debug";
+import Prompt from "prompts";
+
+import { HardhatArtifactResolver } from "../../helpers/hardhat-artifact-resolver.js";
+import { PrettyEventHandler } from "../../helpers/pretty-event-handler.js";
+import { readDeploymentParameters } from "../../helpers/read-deployment-parameters.js";
+import { resolveDeploymentId } from "../../helpers/resolve-deployment-id.js";
+import { bigintReviver } from "../utils/bigintReviver.js";
+import { loadModule } from "../utils/load-module.js";
+import { shouldBeHardhatPluginError } from "../utils/shouldBeHardhatPluginError.js";
+
+interface TaskDeployArguments {
+  modulePath: string;
+  parameters: string;
+  deploymentId: string;
+  defaultSender: string;
+  strategy: string;
+  reset: boolean;
+  verify: boolean;
+  writeLocalhostDeployment: boolean;
+}
+
+const log = debug("hardhat:ignition:task:deploy");
+
+const taskDeploy: NewTaskActionFunction<TaskDeployArguments> = async (
+  {
+    modulePath,
+    parameters: parametersInput,
+    deploymentId: givenDeploymentId,
+    defaultSender,
+    reset,
+    verify,
+    strategy: strategyName,
+    writeLocalhostDeployment,
+  },
+  hre: HardhatRuntimeEnvironment,
+): Promise<DeploymentResult | null> => {
```

Comment:
This was void. I modified it to allow tests to determine if a deployment was a success or not.
This is however, exporting a core type.
One option would be to return a `hardhat-plugin` specific type, or something simpler like boolean.
Another option is to return void and alter the test.

Reply:
i think that's fine